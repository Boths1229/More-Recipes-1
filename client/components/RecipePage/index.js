import React from 'react';
import Header from '../common/header';
import RecipeCardGrid from './recipeCardGrid';
import PopularRecipeCardList from './popularRecipeCardList';



/* eslint-disable no-unused-vars */
/* eslint-disable require-jsdoc */
class RecipePage extends React.Component {

  render() {
    return (
      <div>
        <Header/>
        <div className="content">
          <div className="container cont_area">
            <div className="row">
              <RecipeCardGrid/>
              <PopularRecipeCardList/>
            </div>
          </div>
        </div>
      </div>         
    );
  }
}

export default RecipePage;