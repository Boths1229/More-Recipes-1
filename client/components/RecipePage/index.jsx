/* eslint-disable no-unused-vars */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Header from '../common/header';
import Footer from '../common/footer';
import RecipeCardGrid from './recipeCardGrid';
import PopularRecipeCardList from './popularRecipeCardList';
import { getAllRecipesAction,
  getPopularRecipesAction,
} from '../../actions/recipesAction';

/* eslint-disable require-jsdoc */
/* eslint-disable class-methods-use-this */
class RecipePage extends Component {
  componentWillMount() {
    this.props.actions.getAllRecipesAction();
    this.props.actions.getPopularRecipesAction();
  }

  render() {
    const {
      popularRecipes,
    } = this.props;
    return (
      <div>
        <Header/>
        <div className="content">
          <div className="container cont_area">
            <div className="row">
              <RecipeCardGrid
              allRecipes={this.props.recipes}
              upvoteAction={this.upvoteRecipe}
              />
              <PopularRecipeCardList popularRecipes={popularRecipes}/>
            </div>
          </div>
        </div>
        <Footer/>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    recipes: state.recipes.recipes,
    popularRecipes: state.recipes.popularRecipes,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      getAllRecipesAction,
      getPopularRecipesAction,
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(RecipePage);
