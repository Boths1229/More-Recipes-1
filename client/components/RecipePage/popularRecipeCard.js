import React from 'react';
import img from '../../public/images/recipe-6.jpg'

/* eslint-disable no-unused-vars */
const popularRecipeCard = (props) => {
  return(
    <div className="side_recipes">
      <a className="side_item" href="/recipes/:recipeId">
        <div className="img">
          <img src={img} alt="" style={{height: "auto", width: "100px", marginLeft: "0px", marginTop: "-13.3571px"}}/>
        </div>
        <div className="title dot" style={{wordWrap: "break-word"}}>Yummy Mango Citrus Drink</div>
      </a>
    </div>
  ); 
};

export default popularRecipeCard;