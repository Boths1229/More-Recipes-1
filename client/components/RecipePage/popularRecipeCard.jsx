import React, { PropTypes } from 'react';
import { Link } from 'react-router';
import img from '../../public/images/recipe-6.jpg';

const propTypes = {
  recipeName: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired
};


const popularRecipeCard = ({ recipeName, id }) => (
  <div className="side_recipes">
    <Link className="side_item" to={`/recipes/${id}`}>
      <div className="img">
        <img
          src={img}
          alt=""
          style={{
            height: 'auto',
            width: '100px',
            marginLeft: '0px',
            marginTop: '-13.3571px'
          }}
        />
      </div>
      <div
        className="title dot"
        style={{ wordWrap: 'break-word' }}
      >
        {recipeName}
      </div>
    </Link>
  </div>
);

popularRecipeCard.propTypes = propTypes;

export default popularRecipeCard;
