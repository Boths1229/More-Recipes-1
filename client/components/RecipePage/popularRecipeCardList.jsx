import React, { PropTypes } from 'react';
import PopularRecipeCard from './popularRecipeCard';

const propTypes = {
  popularRecipes: PropTypes.arrayOf(PropTypes.shape()).isRequired
};

const popularRecipeCardList = props => (
  <div className="col-lg-3 col-md-12">
    <aside className="sidebar">
      <div className="side_rec_area">
        <div className="side_title">
          Popular Recipes
        </div>
        {
          props.popularRecipes.map(recipes => (
            <PopularRecipeCard
              recipeName={recipes.recipeName}
              id={recipes.id}
              key={recipes.id}
              img={recipes.image}
            />
          ))
        }
      </div>
    </aside>
  </div>
);

popularRecipeCardList.propTypes = propTypes;

export default popularRecipeCardList;
