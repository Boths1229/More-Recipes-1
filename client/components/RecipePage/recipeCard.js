import React from 'react';
import img from '../../public/images/recipe-5.jpg';
/* eslint-disable no-unused-vars */
const recipeCard = (props) => {
  return(
    <div className="row">
      <div className="col-md-4 card-space">
        <div className="card-deck">
          <article className="card">
            <img src={img} className="card-img-top img-fluid"/>
            <div className="card-body">
              <h3 className="card-title">
                <a href="" id="remove-link">
                  Spicy Chinese Pizza
                </a>
              </h3>
            </div>
            <div >
              <div style={{display: "inline-flex", marginTop: "10px"}}>
                <div>
                  <button className="btn vote-button" type="button">
                    <i className="fa fa-thumbs-up fa-lg" aria-hidden="true"></i>
                  </button>
                  <span style={{marginLeft: "5px", marginRight: "5px"}}>50</span>
                </div>
                <div style={{marginLeft: "20px"}}>
                  <button className="btn vote-button" type="button">
                    <i className="fa fa-thumbs-o-down fa-lg" aria-hidden="true"></i>
                  </button>
                  <span style={{marginLeft: "5px", marginRight:"5px"}}>27</span>
                </div>
              </div>
            </div>
          </article>
        </div>
      </div>
    </div>
  );
  
};


export default recipeCard;