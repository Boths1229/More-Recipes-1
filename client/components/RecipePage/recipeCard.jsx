/* eslint-disable no-unused-vars */
import React from 'react';
import img from '../../public/images/recipe-5.jpg';

const recipeCard = props => (
  <div className='col-md-4 card-space'>
    <div className='card-deck'>
      <article className='card'>
        <img src={img} className='card-img-top img-fluid' />
        <div className='card-body'>
          <h3 className='card-title'>
            <a href="" id="remove-link">
              {props.recipeName}
            </a>
          </h3>
        </div>
        <div >
          <div style={{ display: 'inline-flex', marginTop: '10px' }}>
            <div>
              <button className='btn vote-button' type='button'>
                <i className='fa fa-thumbs-up fa-lg' aria-hidden='true'></i>
              </button>
              <span style={{ marginLeft: '5px', marginRight: '5px' }}>{props.upvotes}</span>
            </div>
            <div style={{ marginLeft: '20px' }}>
              <button className='btn vote-button' type='button'>
                <i className='fa fa-thumbs-o-down fa-lg' aria-hidden='true'></i>
              </button>
              <span style={{ marginLeft: '5px', marginRight: '5px' }}>{props.downvotes}</span>
            </div>
          </div>
        </div>
      </article>
    </div>
  </div>
);


export default recipeCard;
