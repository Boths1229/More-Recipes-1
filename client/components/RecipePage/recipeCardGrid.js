import React from 'react';
import RecipeCard from './recipeCard';


/* eslint-disable no-unused-vars */
const recipeCardGrid = (props) => {
  return(
    <div className="col-lg-9 col-md-12">
      <div className="row">
        <div className="col-md-12">
          <div className="line small">
            <h2>Recipes</h2>
          </div>
        </div>

      </div>
      <RecipeCard/>
    </div>
  );
  
};

export default recipeCardGrid;
