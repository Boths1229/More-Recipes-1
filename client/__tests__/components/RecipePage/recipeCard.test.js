import React from 'react';
import { mount } from 'enzyme';
import { RecipeCard } from '../../../components/RecipePage/recipeCard';

let props;
const err = {
  response: {
    data: {
      message: 'Session has expired'
    }
  }
};
const setup = () => {
  props = {
    actions: {
      voteRecipeAction: () => Promise.reject(err),
      changeAuthAction: () => {},
    },
    id: 1,
    i: 1,
    recipeName: 'Rice',
    upvotes: 3,
    downvotes: 0,
    image: 'image',
    user: 'tywo'
  };

  return mount(<RecipeCard {...props} />, { context: {} });
};

describe('Recipe card component', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should toast error for upvoting recipe when session expired', () => {
    const wrapper = setup();
    const action = wrapper.instance();
    const spy = jest.spyOn(wrapper.instance(), 'upvoteRecipe');
    action.upvoteRecipe({ preventDefault: () => {} });
    expect(spy).toBeCalled();
  });

  it('should toast error for downpvoting recipe when session expired', () => {
    const wrapper = setup();
    const action = wrapper.instance();
    const spy = jest.spyOn(wrapper.instance(), 'downvoteRecipe');
    action.downvoteRecipe({ preventDefault: () => {} });
    expect(spy).toBeCalled();
  });

  it('should toast error for upvoting recipe when user is not logged in', () => {
    const wrapper = setup();
    const err = {
      response: {
        data: {
          message: 'not logged in'
        }
      }
    };
    wrapper.setProps({
      actions: { voteRecipeAction: () => Promise.reject(err) }
    });
    const action = wrapper.instance();
    const spy = jest.spyOn(wrapper.instance(), 'upvoteRecipe');
    action.upvoteRecipe({ preventDefault: () => {} });
    expect(spy).toBeCalled();
  });

  it('should toast error for downvoting recipe when user is not logged in', () => {
    const wrapper = setup();
    const err = {
      response: {
        data: {
          message: 'not logged in'
        }
      }
    };
    wrapper.setProps({
      actions: { voteRecipeAction: () => Promise.reject(err) }
    });
    const action = wrapper.instance();
    const spy = jest.spyOn(wrapper.instance(), 'downvoteRecipe');
    action.downvoteRecipe({ preventDefault: () => {} });
    expect(spy).toBeCalled();
  });
});
