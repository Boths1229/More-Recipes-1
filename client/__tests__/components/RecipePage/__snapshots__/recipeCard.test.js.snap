// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe card component should render correctly 1`] = `
<RecipeCard
  actions={
    Object {
      "changeAuthAction": [Function],
      "voteRecipeAction": [Function],
    }
  }
  downvotes={0}
  i={1}
  id={1}
  image="image"
  recipeName="Rice"
  upvotes={3}
  user="tywo"
>
  <div
    className="col-md-4 card-space"
  >
    <div
      className="card-deck"
    >
      <article
        className="card recipe-card"
      >
        <img
          alt="recipe"
          className="card-img-top img-fluid"
          src="image"
          style={
            Object {
              "height": "150px",
            }
          }
        />
        <div
          className="card-body"
        >
          <h3
            className="card-title"
          >
            <Link
              id="remove-link"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/recipes/1"
            >
              <a
                id="remove-link"
                onClick={[Function]}
                style={Object {}}
              >
                Rice
              </a>
            </Link>
          </h3>
        </div>
        <div>
          <div
            style={
              Object {
                "display": "inline-flex",
                "marginTop": "10px",
              }
            }
          >
            <div>
              <button
                className="btn vote-button"
                data-for="upvote"
                data-tip={true}
                id={1}
                onClick={[Function]}
                type="button"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-up"
                >
                  <span
                    style={
                      Object {
                        "marginLeft": "5px",
                        "marginRight": "5px",
                      }
                    }
                  >
                    3
                  </span>
                </i>
              </button>
              <ReactTooltip
                id="upvote"
                insecure={true}
                place="top"
                resizeHide={true}
                type="dark"
                wrapper="div"
              >
                <div
                  className="__react_component_tooltip place-top type-dark "
                  data-id="tooltip"
                />
              </ReactTooltip>
            </div>
            <div
              style={
                Object {
                  "marginLeft": "20px",
                }
              }
            >
              <button
                className="btn vote-button"
                data-for="downvote"
                data-tip={true}
                onClick={[Function]}
                type="button"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-thumbs-o-down"
                >
                  <span
                    style={
                      Object {
                        "marginLeft": "5px",
                        "marginRight": "5px",
                      }
                    }
                  >
                    0
                  </span>
                </i>
              </button>
            </div>
            <ReactTooltip
              id="downvote"
              insecure={true}
              place="right"
              resizeHide={true}
              type="dark"
              wrapper="div"
            >
              <div
                className="__react_component_tooltip place-top type-dark "
                data-id="tooltip"
              />
            </ReactTooltip>
          </div>
        </div>
        <div
          className="card-footer text-muted"
          style={
            Object {
              "fontSize": "12px",
              "marginTop": "15px",
            }
          }
        >
          recipe by: 
          tywo
        </div>
      </article>
    </div>
  </div>
</RecipeCard>
`;
